### PR TITLE
[TRTLLM-14554][perf] Batch Wan VAE decode along the temporal axis

### PR DESCRIPTION
**Pull request by taianz-nv**
## Dev Engineer Review

- Added temporal batching for Wan VAE decoding.
- Preserved first-frame causal convolution cache initialization.
- Added dtype- and parallel-size-based chunk selection with environment overrides.
- Preserved backward compatibility with `temporal_chunk_size=1`.
- Added validation for invalid chunk sizes and fallback behavior.
- Reduced decoder invocations and added parity checks.
- No dependency or unrelated configuration changes identified.

## QA Engineer Review

- Added tests for `_decode_chunk_slices`.
- Added tests for `_native_decode_chunk_size`.
- Updated Wan2.2 residual and Wan2.1 non-residual parity tests.
- Added Wan2.2 consistency coverage for chunk sizes 1 and 4.
- These are test-code changes outside `tests/integration/test_lists/`.
- No corresponding `test-db/` or `qa/` coverage was identified.
- Verdict: needs follow-up.

## Description

Wan VAE decode currently invokes the complete decoder once for every latent
frame. Each invocation walks the same spatial decoder hierarchy and launches
its convolutions, normalization, activation, layout, and spatial-halo
operations. Causal Conv3d can instead process several adjacent temporal
positions together while retaining the same preceding feature context.

This PR keeps the first latent frame separate to initialize the causal feature
cache, then groups later frames into contiguous temporal batches:

```text
chunk size 1: [0] [1] [2] ... [20]            -> 21 decoder invocations
chunk size 4: [0] [1,2,3,4] ... [17,18,19,20] ->  6 decoder invocations
```

It is called **temporal decode batching** because the grouping is exclusively
along tensor dimension `D/T`, the latent time/frame axis. It does not batch
requests, images, channels, height, or width. Spatial H/W still changes only
between decoder stages in the usual upsampling hierarchy.

## Chunk-size selection

The native parallel wrapper selects the chunk size from `(parallel_size,
tensor dtype)`. The initial tuned table contains four-rank BF16 -> 4. One rank
keeps the existing chunk size 1 behavior. Other multi-rank/dtype combinations
use a conservative chunk size 2 until follow-up sweeps add tuned entries.

`TLLM_WAN_VAE_DECODE_TEMPORAL_CHUNK_SIZE=<positive integer>` overrides the selection
for benchmark sweeps. This is an internal tuning switch rather than a new
public API/configuration field. The Diffusers wrapper retains chunk size 1.

## Why causal correctness is preserved

Each temporal chunk traverses the complete decoder. The first frame is decoded
alone to initialize every causal Conv3d cache. Later slices batch adjacent
frames through the complete decoder while reusing the preceding slice's
per-layer temporal context. Each pass resets the layer cursor to layer zero,
but the per-layer feature-cache array persists across slices. Only complete
decoder outputs are concatenated along the temporal dimension afterward.

## Performance

For the 21-frame latent in the four-GPU NSC native-BF16 VAE benchmark, chunk
size 4 reduces complete decoder invocations from 21 to 6 (`71.4%` fewer).
In the same selected-stack A/B sweep, chunk size 4 measured `590.430 ms`, versus
`603.523 ms` for chunk size 3, saving another `13.093 ms` (`2.2%`). The
single-frame baseline from the corresponding stable campaign was `634.365 ms`.

One-GPU temporal batching regressed in the validation sweep, so its automatic
selection remains 1. Quantized implementations are intentionally outside this
PR: if their public tensors retain BF16 while the operator computes in another
format, their follow-up integration should provide an explicit mode key rather
than infer the mode from `z.dtype`. The environment override already supports
collecting those tuning sweeps.

## Test Coverage

- Added exact slice-schedule coverage, including zero frames, the first-frame
  singleton, and a short final batch.
- Added validation for zero and negative chunk sizes, tuned/fallback selection,
  and valid/invalid environment overrides.
- Wan2.2 360p/720p and Wan2.1 checkpoint decode parity against Diffusers now
  exercise chunk size 4.
- Added direct Wan2.2 checkpoint parity between framewise chunk size 1 and
  temporal chunk size 4.
- NSC job `1559025` passed direct chunk1-vs-chunk4 checkpoint validation:
  `max_abs=0.001325`, `relative_mean=0.000187`.
- Existing NSC checkpoint parity and stable four-GPU BF16 chunk sweeps pass.
- Repository pre-commit hooks, Python compilation, `git diff --check`, DCO,
  title, checklist, and API compatibility checks pass.

## PR Checklist

- [x] The change is independent of FP4.
- [x] Single-rank decode retains the existing chunk size 1 behavior.
- [x] Untuned multi-rank/dtype combinations use a conservative fallback.
- [x] The first frame remains separate for causal-cache initialization.
- [x] No public API, dependency, CODEOWNERS, or architecture-diagram changes.

## FP4 integration report

This precision-independent optimization composes with the FP4 Wan VAE integration in #17262. The consolidated performance/accuracy report and dependency map are maintained there.

### CODE REVIEW

File: tests/unittest/_torch/visual_gen/test_wan_vae_temporal_batching.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+
+@pytest.mark.parametrize("override", ["0", "-1", "invalid"])
+def test_native_decode_chunk_size_rejects_invalid_env_override(
+    monkeypatch,
+    override: str,
+) -> None:
+    monkeypatch.setenv(TRTLLM_WAN_VAE_DECODE_CHUNK_SIZE, override)
+    with pytest.raises(ValueError, match="must be a positive integer"):
+        _native_decode_chunk_size(4, torch.bfloat16)
```
[the remark this answers is not in the document]
Comment (taianz-nv) [reply tests/unittest/_torch/visual_gen/test_wan_vae_temporal_batching.py]: Addressed in c10154db7e. The new temporal-batching test file is now registered in `tests/integration/test_lists/test-db/l0_b200.yml`; `scripts/test_to_stage_mapping.py` maps it to the DGX B200 PyTorch stages, and the targeted pre-commit/test-list validation passes.

File: tensorrt_llm/_torch/visual_gen/models/wan/parallel_vae.py [hunk 2]
```diff
@@ -33,6 +34,40 @@
 )
 from tensorrt_llm._torch.visual_gen.utils import as_tuple
 
+TRTLLM_WAN_VAE_DECODE_CHUNK_SIZE = "TRTLLM_WAN_VAE_DECODE_CHUNK_SIZE"
```
Comment (zhenhuaw-me) [inline tensorrt_llm/_torch/visual_gen/models/wan/parallel_vae.py]:
```suggestion
TRTLLM_WAN_VAE_DECODE_CHUNK_SIZE = "TLLM_WAN_VAE_DECODE_TEMPORAL_CHUNK_SIZE"
```

File: tests/unittest/_torch/visual_gen/test_wan_vae_temporal_batching.py [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Unit tests for native Wan VAE temporal decode batching."""
+
+import pytest
+import torch
+
+from tensorrt_llm._torch.visual_gen.models.wan.parallel_vae import (
```
Comment (zhenhuaw-me) [inline tests/unittest/_torch/visual_gen/test_wan_vae_temporal_batching.py]: kinda overkill to test this and it's cpu test

File: tensorrt_llm/_torch/visual_gen/models/wan/parallel_vae.py [hunk 4]
```diff
@@ -33,6 +34,40 @@
...
+
+# Keep tuned entries explicit so future sweeps can extend this table by
+# parallel size and tensor dtype without changing the selection policy.
+_NATIVE_DECODE_CHUNK_SIZES: dict[tuple[int, torch.dtype], int] = {
+    (4, torch.bfloat16): 4,
+}
+_DEFAULT_MULTI_GPU_DECODE_CHUNK_SIZE = 2
+
+
+def _native_decode_chunk_size(
```
Comment (zhenhuaw-me) [inline tensorrt_llm/_torch/visual_gen/models/wan/parallel_vae.py:47]: better to doc why need this function

**Review by zhenhuaw-me — APPROVED** [review]

**Review by fredricz-20070104 — APPROVED** [review]
## Review summary - Approve

Reviewed the full diff; no blocking or major issues found.

Minor, non-blocking notes:
- `tests/unittest/_torch/visual_gen/test_wan_vae.py`: monkeypatch params lack type annotations
- `tests/unittest/_torch/visual_gen/test_wan_vae.py`: New tests not wired into CI/QA test lists

_Automated review by NVCortex Lite, run by @fredricz-20070104._

**Review by tburt-nv — APPROVED** [review]
Confirmed unittest/_torch/visual_gen/test_wan_vae.py is scheduled in CI.

File: tests/unittest/_torch/visual_gen/test_wan_vae_temporal_batching.py (hunk 3 above)
Comment (taianz-nv) [reply tests/unittest/_torch/visual_gen/test_wan_vae_temporal_batching.py]: Removed from gpu test list

File: tests/unittest/_torch/visual_gen/test_wan_vae.py [hunk 5]
```diff
@@ -114,6 +122,67 @@ def test_use_diffuser_vae_env_zero_keeps_native(monkeypatch):
...
+
+
+@pytest.mark.parametrize("override", ["0", "-1", "invalid"])
+def test_native_decode_chunk_size_rejects_invalid_env_override(
+    monkeypatch,
+    override: str,
+) -> None:
+    monkeypatch.setenv(TLLM_WAN_VAE_DECODE_TEMPORAL_CHUNK_SIZE, override)
+    with pytest.raises(ValueError, match="must be a positive integer"):
+        _native_decode_chunk_size(4, torch.bfloat16)
```
[the remark this answers is not in the document]
Comment (taianz-nv) [reply tests/unittest/_torch/visual_gen/test_wan_vae.py:185]: Addressed in bbe78d1d10. The three new monkeypatch fixtures are now typed as `pytest.MonkeyPatch`, and the environment-override and chunk-parity tests have explicit `-> None` return annotations. The file passes the targeted pre-commit hooks.

**Review by BowenFu — APPROVED** [review]
Verified the regression axis: `temporal_chunk_size` defaults to 1 everywhere, `_decode_chunk_slices(n, 1)` reproduces the old `range(num_frame)` partition exactly, and only `ParallelVAE_TrtllmWan` injects a >1 value (`world_size<=1` still returns 1), so single-GPU and the diffusers-backed `ParallelVAE_Wan` are unchanged.

Two non-blocking notes: peak VAE-decode activation memory now scales with chunk size on the multi-GPU path with no fallback if it doesn't fit; and the two diffusers-parity tests were switched to `temporal_chunk_size=4`, so the shipped chunk=1 default is only covered transitively via `test_wan22_temporal_chunk4_matches_chunk1_checkpoint`.
